src/app/components/header/header.component.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnInit,
+  ViewEncapsulation,
+} from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map, switchMap, take } from 'rxjs/operators';
+import { CategoryQueryModel } from '../../query-models/category.query-model';
+import { InMemoryCategoriesStorage } from '../../storages/categories/in-memory-categories.storage';
+import { CategoriesService } from '../../services/categories.service';
+import { IsVisibleQueryModel } from '../../query-models/is-visible.query-model';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HeaderComponent implements OnInit {
+  readonly categories$: Observable<CategoryQueryModel[]> =
+    this._categoriesStorage.select().pipe(
+      map((categories) =>
+        categories.map((category) => ({
+          name: category.name,
+          link: `categories/${category.id}`,
+        }))
+      )
+    );
+  private _isNavbarVisibleSubject: BehaviorSubject<IsVisibleQueryModel> =
+    new BehaviorSubject<IsVisibleQueryModel>({ isNavbarVisible: false });
+  public isNavbarVisible$: Observable<IsVisibleQueryModel> =
+    this._isNavbarVisibleSubject.asObservable();
+
+  constructor(
+    private _categoriesStorage: InMemoryCategoriesStorage,
+    private _categoriesService: CategoriesService
+  ) {}
+
+  ngOnInit() {
+    this._categoriesService
+      .getAll()
+      .pipe(
+        switchMap((categories) => this._categoriesStorage.set(categories)),
```

Comment:
You set this storage in two places. Choose one and do this there. With your solution you subscribe to categoriesSerbice.getAll() twice so maybe storage is not necessary?  Choose one place where you do this, so that you can use this data in all places